dojo_plugin/models/__init__.py

```diff
@@ -708,6 +708,31 @@ class Emojis(Awards):
     __mapper_args__ = {"polymorphic_identity": "emoji"}
 
 
+# CTFd already has a "Tokens" model that has a variable type field, but the
+#  authentication logic doesn't care what type the token is, only if it is valid.
+# So if we were to re-use that model, it would require moneky-patching security-critical
+#  code and breaking the CTFd authors' assumptions, which makes me nervous. So, to be
+#  extra safe, we are going to define a new model. 
+class WorkspaceTokens(db.Model):
+    __tablename__ = "workspace_tokens"
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey("users.id", ondelete="CASCADE"))
+    created = db.Column(db.DateTime, default=datetime.datetime.utcnow)
+    expiration = db.Column(
+        db.DateTime,
+        default=lambda: datetime.datetime.utcnow() + datetime.timedelta(days=30),
+    )
+    value = db.Column(db.String(128), unique=True)
```

Comment:
Instead of a generic value, we should probably have a `authorizee_user_id` (and maybe rename `user_id` to `authorizer_user_id`?) This should also be a ForeignKey to a users.id, delete on cascade, etc.

Reply:
Also, I absolutely agree with your thinking around not subtyping the already-existing CTFd tokens since it's baked into security-critical paths.

Reply:
> Instead of a generic value
the value field holds the token itself. are you saying we should replace sharing a token string with specifying the username you want to authorize, kinda like github repo invitations? or are you saying we should we keep the token but limit it to a certain user?

Reply:
To further clarify, I was initially envisioning sharing with a specific user, but I think the UX is much simpler by embedding that "authorization" into the URL itself (i.e. if you have the URL, you *are* authorized) as I realized in my most recent comment.